nemo/src/io/formats/sparql.rs

```diff
@@ -125,21 +123,23 @@ impl FormatBuilder for SparqlBuilder {
         // Create and return a new ResourceSpec to replace the existing ResourceSpec
         Ok(Self {
             value_formats,
-            limit,
             endpoint,
             query,
         })
     }
 
-    // TODO: do we need to parse parameters here again? that would be annoying
-    fn override_resource_spec(&self, direction: Direction) -> Option<ResourceSpec> {
+    fn override_resource_builder(&self, direction: Direction) -> Option<ResourceBuilder> {
         if matches!(direction, Direction::Export) {
             None
         } else {
-            Some(ResourceSpec::from_endpoint(
-                self.endpoint.clone(),
-                self.query.to_sse(),
-            ))
+            let mut resource_builder = ResourceBuilder::from_valid_iri(self.endpoint.clone());
+            let query = self.query.to_string();
+            if query.len() > HTTP_GET_CHAR_LIMIT {
+                resource_builder.parameter_mut(HttpParamType::Post, String::from("query"), query);
+            } else {
+                resource_builder.parameter_mut(HttpParamType::Get, String::from("query"), query);
+            }
+            Some(resource_builder)
```

Comment:
Would be cleaner (and safer, should `Direction` ever get another variant) to have a `match direction { … }` here